google_drive/google_calendar.py

```diff
@@ -35,62 +37,85 @@ class GoogleCalendarBlock(XBlock):
-                mozallowfullscreen="true"
-                webkitallowfullscreen="true">
-            </iframe>
-        """))
-
-    def student_view(self, context={}):
+        default=1
+    )
+
+    views = [(0, 'Week'), (1, 'Month'), (2, 'Agenda')]
+
+    # Context argument is specified for xblocks, but we are not using herein
+    def student_view(self, context):  # pylint: disable=unused-argument
         """
         Player view, displayed to the student
         """
-
         fragment = Fragment()
-        context.update({
+
+        view = self.views[self.default_view][1]
+        iframe = CALENDAR_IFRAME.format(view, self.calendar_id, self.display_name)
+
+        fragment.add_content(RESOURCE_LOADER.render_template(CALENDAR_TEMPLATE, {
             "self": self,
-        })
-        fragment.add_content(render_template('/templates/html/google_docs.html', context))
-        fragment.add_css(load_resource('public/css/google_docs.css'))
-        fragment.add_javascript(load_resource('public/js/google_docs.js'))
+            "iframe": iframe
+        }))
+        fragment.add_css(RESOURCE_LOADER.load_unicode('public/css/google_calendar.css'))
+        fragment.add_javascript(RESOURCE_LOADER.load_unicode('public/js/google_calendar.js'))
 
-        fragment.initialize_js('GoogleDocumentBlock')
+        fragment.initialize_js('GoogleCalendarBlock')
 
         return fragment
 
-    def studio_view(self, context):
+    # Context argument is specified for xblocks, but we are not using herein
+    def studio_view(self, context):  # pylint: disable=unused-argument
         """
         Editing view in Studio
         """
         fragment = Fragment()
-        fragment.add_content(render_template('/templates/html/google_docs_edit.html', {
+        # Need to access protected members of fields to get their default value
+        fragment.add_content(RESOURCE_LOADER.render_template(CALENDAR_EDIT_TEMPLATE, {
             'self': self,
+            'defaultName': self.fields['display_name']._default,  # pylint: disable=protected-access
+            'defaultID': self.fields['calendar_id']._default  # pylint: disable=protected-access
         }))
-        fragment.add_javascript(load_resource('public/js/google_docs_edit.js'))
+        fragment.add_javascript(RESOURCE_LOADER.load_unicode('public/js/google_calendar_edit.js'))
+        fragment.add_css(RESOURCE_LOADER.load_unicode('public/css/google_edit.css'))
 
-        fragment.initialize_js('GoogleDocumentEditBlock')
+        fragment.initialize_js('GoogleCalendarEditBlock')
 
         return fragment
 
+    # suffix argument is specified for xblocks, but we are not using herein
     @XBlock.json_handler
-    def studio_submit(self, submissions, suffix=''):
-
+    def studio_submit(self, submissions, suffix=''):  # pylint: disable=unused-argument
+        """
+        Change the settings for this XBlock given by the Studio user
+        """
         self.display_name = submissions['display_name']
```

Comment:
We should probably do something about this in the XBlock API, but you should handle failures here: submissions might not be a dict, and it might not have the keys you expect.  Not sure why, but the Schoolyourself XBlock has a bug written against it where it gets strings sometimes instead of dicts?